DesignDoc/RMA.md

```diff
@@ -1,3 +1,205 @@
 # Request Merchandise Authorization (RMA) notification structure design
 
-## TBA
+## Intent
+
+Structure for RMA process.
+
+## Definitions
+
+- Consumer : Customers of device vendors (Eg: Google, Microsoft, X).
+- Publisher: Device vendor (Ciena, Adva, Infinera, X).
+
+## Features 
+
+RMA case is raised by the consumer to request the replacement of the defective parts. It consists of the following subflows:
+1. Consumer starts to request the RMA from the Publisher, and wait for the review/approval from the publisher
+1. Publisher ships the replacement, and notify the consumer about the shipment status of the replacement
+1. Consumer request techincial dispatch/schedule if needed
+1. Consumer ships the defective part for failure analysis, and notify the consumer about the shipment status of the defective parts
+1. Publisher notifies the failure analysis report to consumer
+## Out of scope
+
+We would not be discussing the transport methodologies preferred for this notification message.
+
+## Stages of RMA Lifecycle
+
+### Creation
+
+At this stage RMA(s) is raised by the Consumer, this might include:
+
+1. Collect the information, including 
+   * device/part basic information, e.g., serial number, part type, OS software version
+   * device/part alarms and logs
+   * site information where the defective parts resides
+   * contact information from Consumer side
+   * RMA service level, e.g., normal / escalate service
+
+1. Consumer request for RMA from Publisher
+1. Request type for each customer (email, API, manual).
+
+### Review/Approval 
+
+At this stage, Publisher will validate the required information for the RMA case, and approve / reject / request more information, which may include:
+   * whether any mandotary information is missing
+   * whether a kind of service level is supported for specific region
+
+### Parts tracking 
+
+There are two seperate flows at this stage:
+   1. Replacement shippment:
+      1. Publisher will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+   2. Defective part shippment:
+      1. Consumer will provide the shippment information, including carrier, planned arrive date 
+      1. Publisher will notify the status of the replacement shippment and delivery
+
+### Notification
+
+One RMA case is dedicated to one part with one unique identifier, e.g., serial number. Each RMA involves multiple communication messages between the publisher and consumer. Each RMA case has a unique ID.
+
+## Notification structure
+
+### General format
+
+`field_name:field_type:description`
+
+### Field Details
+
+#### case_id
+
+Identifier for a RMA case which dedicated to a part with a unique identifier. 
```

Comment:
Identifier for a RMA case dedicated to a part with a unique identifier. 

The unique identifier is for the case or the RMA part?

Reply:
it is for RMA case. Fixed.